src/js/helpers/Util.js

```diff
@@ -113,6 +113,26 @@ var Util = {
 
     return obj;
   },
+  objectCreateWithAdder: function (valuePrefix = "") {
+    const obj = {};
+
+    // Make the properties of the object read-only, but the object extensible.
+    Object.defineProperty(obj, "add", {
+      enumerable: false,
+      configurable: false,
+      writable: false,
+      value: function (key) {
+        Object.defineProperty(obj, key, {
+          enumerable: false,
+          configurable: false,
+          writable: false,
+          value: valuePrefix + key
```

Comment:
As you've added this method to create protected objects for plugins that ensure that every action, event or mount point key is unique you should use Symbols instead of strings as values. You should also rename this utils and maybe move it to a different file as it's specifically added for the plugins and probably isn't useful in any other case and the name is kinda strange and unclear.